contracts/genie-airdrop/src/contract.rs

```diff
@@ -249,11 +260,15 @@ pub fn handle_claim(
 
     // Transfer assets to the recipient
     let config = &CONFIG.load(deps.storage)?;
-    let messages = vec![build_transfer_asset_msg(
-        recipient,
-        &config.asset,
-        claim_amount,
-    )?];
+    let messages = if claim_amount == Uint128::zero() {
```

Comment:
just a question: does it make sense to not even call the smart contract if there's nothing to claim?
i.e some pre processing on the backend first, after generating the raffle tickets?

Although I guess that means the user won't be prompted to sign the claim on the frontend which is weird too 🤔